test/hooks/index.js

```diff
@@ -121,4 +279,325 @@ describe('hooks', () => {
 				assert.ok(wasCalled);
 			});
 	});
+
+	it('supports asset emission', () => {
+		return rollup
+			.rollup({
+				input: 'input',
+				experimentalCodeSplitting: true,
+				experimentalDynamicImport: true,
+				plugins: [
+					loader({ input: '' }),
+					{
+						transform (code, id) {
+							const assetId = this.emitAsset('test.ext', 'hello world');
+							return `export default import.meta.ROLLUP_ASSET_URL_${assetId};`;
+						}
+					}
+				]
+			})
+			.then(bundle => {
+				return bundle.generate({ format: 'es' });
+			})
+			.then(outputBundle => {
+				assert.equal(outputBundle['assets/test-19916f7d.ext'], 'hello world');
+				assert.equal(outputBundle['input.js'].code, `var input = new URL(\'../assets/test-19916f7d.ext\', import.meta.url).href;\n\nexport default input;\n`);
+			});
+	});
+
+	it('supports CommonJS asset urls', () => {
```

Comment:
I tried this for `amd` and got something like

```javascript
define(function () { 'use strict';

	console.log(new URL((typeof process !== 'undefined' && process.versions && process.versions.node ? 'file:' : '') + module.uri + '/../assets/asset.ext').href);

});
```

I would have expected `module` to be imported by the amd loader similar to how it is done for `import.meta.url`.

Also, `system` is not supported at all as a format but throws in a rather non-pleasing way with a TypeError.

Reply:
Thanks, will fix.